velox/connectors/hive/HiveConnector.h

```diff
@@ -41,18 +41,23 @@ class HiveConnector : public Connector {
           std::string,
           std::shared_ptr<connector::ColumnHandle>>& columnHandles,
       ConnectorQueryCtx* connectorQueryCtx) override {
+    dwio::common::ReaderOptions options(connectorQueryCtx->memoryPool());
+    options.setMaxCoalesceBytes(
+        HiveConfig::maxCoalescedBytes(connectorQueryCtx->config()));
+    options.setMaxCoalesceDistance(
```

Comment:
Would it make sense to have a helper functions that creates a ReaderOptions based on a ConnectorQueryCtx? I'm concerned about having to manually carry each parameter every time we need to create a ReaderOptions (I found at least other two places where we needed this; one in Prestissimo, another in an internal project).

Reply:
I agree but wanted to follow the "Status Quo". In my past lives, configuration was handled with a global singleton factory and you could retrieve an immutable view or a mutable copy of the configuration from anywhere the code. To get specific options i.e. ReaderOptions, the API would accept a string that identified the options by a name. You would register your options like any other factory pattern. I have never liked "passing" options around and found it easier to maintain long term if options/config are centralized. Depending on needs you can also expose the original options as a mutable object but that should be used in only one place per option type i.e. ReaderOptions where maybe gflags from a component are parsed and stored for global access later. Right now, I am not sure how to handle this correctly for just one type of option like ReaderOptions. It seems like more design may be needed?